subnet/license.go

```diff
@@ -170,6 +172,36 @@ func (lv *LicenseValidator) ValidateLicense() (*licverifier.LicenseInfo, error)
 	return lv.ParseLicense(string(licData))
 }
 
+func getDurationForNextLicenseCheck(li *licverifier.LicenseInfo) time.Duration {
+	if li.ExpiresAt.Before(time.Now()) {
+		// expired, within grace period. schedule daily
+		return time.Hour * 24
+	}
+	// not expired, schedule to check just after expiry
+	return time.Until(li.ExpiresAt.Add(time.Second))
+}
+
+func (lv *LicenseValidator) scheduleNextLicenseCheck(li *licverifier.LicenseInfo, acceptedPlans []string) error {
+	duration := getDurationForNextLicenseCheck(li)
+	timer := time.NewTimer(duration)
+	defer timer.Stop()
+
+	ctxt := context.Background()
+	for {
+		select {
+		case <-timer.C:
+			li, err := lv.ValidateEnterpriseLicense(acceptedPlans)
+			if err != nil {
+				fmt.Println("license validation failed:", err.Error())
```

Comment:
Just return error, no printing needed and do not exit here.

Reply:
@harshavardhana exiting directly from here because this is happening in background, so can't just return error. And we want enterprise tools to exit when license has expired. Do you want me to implement async communication with the app (using channels?) and let the application code decide how to exit on receiving license expired message?

Reply:
@harshavardhana I've modified the code to return the error through a license expiry channel. PTAL.